domain/batch.js

```diff
@@ -22,8 +22,16 @@ class Batch {
         this.filePrefix = filePrefix;
     }
 
-    addFile(name, size, signedURL) {
-        const file = new BatchFile(name, size, signedURL, this.filePrefix);
+    addDataFile(name, size){
+        this.#addFile(name, size, null, true)
+    }
+
+    addMetadataFile(name, size, signedURL){
+        this.#addFile(name, size, signedURL, false)
+    }
+
+    #addFile(name, size, signedURL, isDataFile) {
```

Comment:
can we here signedURL to identify the nodeType instead? I think two dataFile/metadataFile function work as well too.